src/foam/dao/index/Plan.js

```diff
@@ -166,3 +166,79 @@ foam.CLASS({
     }
   ]
 });
+
+/**
+  Merges results from multiple sub-plans and deduplicates, sorts, and
+  filters the results.
+
+  TODO: account for result sorting in cost?
+*/
+foam.CLASS({
+  package: 'foam.dao.index',
+  name: 'MergePlan',
+  extends: 'foam.dao.index.AltPlan',
+
+  requires: [
+    'foam.dao.DedupSink',
+    'foam.dao.LimitedSink',
+    'foam.dao.SkipSink',
+    'foam.dao.OrderedSink',
+  ],
+
+  methods: [
+    /**
+      Executes sub-plans, limiting results from each, then merges results,
+      removes duplicates, sorts, skips, and limits.
+    */
+    function execute(promise, sink, skip, limit, order, predicate) {
+      // TODO: Investigate pre-sorted results from subqueries being
+      //   zipped together quickly
+
+      var resultSink = this.DedupSink.create({
+        delegate: this.decorateSink_(sink, skip, limit, order)
```

Comment:
This is going to be very inefficient when used with skip and limit since it will require visiting and the resorting all data, not just the data in the skip->skip+limit range. The data is already sorted properly, so you shouldn't need to sort it again. If instead of executing the subPlans in series, you instead execute them all in parallel and merge the results, then you won't need to resort or visit any data after skip+limit.